hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Hangman.Game do
+    defstruct(
+        turns: 7,
+        game_state: :initializing,
+        used: %MapSet{},
+        word: [],
+        last_guess: "",
+    )
+
+  def new_game() do
+    word = String.codepoints( Dictionary.random_word() )
+    word = Enum.drop word, -1
+    %Hangman.Game{ word: word }
+  end
+
+  def tally(game = %Hangman.Game{}) do
+    letters = word_mask(game.word,game.used)
+    %{letters: letters,
+     turns_left: game.turns,
+     game_state: game.game_state,
+     used: Enum.sort( MapSet.to_list(game.used) ),
+     last_guess: game.last_guess}
+  end
+
+  def make_move(game = %Hangman.Game{}, guess) do
```

Comment:
### -6 idiomatic

First, I appreciate the way you avoided conditional code. In particular, the turnDecrement parameter `in_word || in_used1 is cute.

But this is still a long, hard to read (and hence hard to change) linear function.

Could you split it into maybe three of four functions which implement a kind of decision tree, thereby localizing a lot of this code.


Also, there's a naming inconsistency: `turnDecrement` vs `turn_decremenent`